forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -512,26 +539,22 @@ private Set<Hash> commitsWithTitle(String commitTitle, Map<String, Set<Hash>> co
         return Set.of();
     }
 
-    private Hash commitWithComment(String commitTitle,
-                                   ZonedDateTime commentCreatedAt,
-                                   HostUser author,
-                                   Map<String, Set<Hash>> commitTitleToCommits) {
+    private Optional<CommitComment> findComment(String commitTitle,
+            String commentId,
+            Map<String, Set<Hash>> commitTitleToCommits) {
         var candidates = commitsWithTitle(commitTitle, commitTitleToCommits);
-        if (candidates.size() == 1) {
-            return candidates.iterator().next();
-        }
-
-        for (var candidate : candidates) {
-            var comments = commitComments(candidate);
-            for (var comment : comments) {
-                if (comment.createdAt().equals(commentCreatedAt) &&
-                    comment.author().equals(author)) {
-                    return candidate;
-                }
-            }
+        // Even if there is only one candidate, we need to make sure the comment
+        // exists on that commit before we try to process it. If this fails it's
+        // most likely due to inconsistent data from GitLab, which should
+        // eventually clear up on subsequent tries.
+        Optional<CommitComment> found = candidates.stream()
+                .flatMap(candidate -> commitComments(candidate).stream())
+                .filter(comment -> comment.id().equals(commentId))
+                .findFirst();
+        if (found.isEmpty()) {
+            log.warning("Did not find commit with title " + commitTitle + " for repository " + projectName);
         }
-
-        throw new RuntimeException("Did not find commit with title " + commitTitle + " for repository " + projectName);
+        return found;
```

Comment:
Just want to confirm that if we stop throwing exceptions here, the commitComment will be handled again in the next CommitCommentsWorkItem.